rootmap/sqlrootmap.go

```diff
@@ -19,38 +20,173 @@ SQLRootmap is a rootmap implementation backed by a sql.DB. The only database
 that has been used or tested is sqlite.
 */
 
+type sqlVersionStore struct {
+	reservationSize int
+	max             uint64
+	initialized     bool
+	mtx             sync.Mutex
+	c               uint64
+}
+
 type sqlRootmap struct {
 	db  *sql.DB
 	mtx *sync.Mutex
+	vs  *sqlVersionStore
```

Comment:
I think that if we want to keep this isolation, we need to migrate the methods that reference the fields of versionStore (like NextVersion) back under the versionStore, and have the rootmap delegate them. I think you can handle that without explicit delegation routines using struct embedding -

```
type sqlRootmap struct {
	db  *sql.DB
	mtx *sync.Mutex
	*sqlVersionStore
Comment

	tables map[table]bool
}
```

as we currently have this, "sqlVersionStore" has been reduced to a simple data container that is manipulated by the rootMap. I feel that looking at this, having the rootmap own the versionStore seems like a nice segregation but we should probably complete that by un-smushing the methods.

Once that's done versionStore can also go in its own file in this folder.

Reply:
Should be done!